### ISSUE TITLE
YouTube music listening activity

### ISSUE THREAD

**Issue by mudrhiod** [vendetta-mod/ideas#213]
### Discord username

Moodle.v2

### Plugin description

The same as discord's official Spotify activity but for YouTube music.

### Elaborate

Shows off the song you're listening to on YouTube music as a rich presence, with the song name, album art, time elapsed etc

~~I don't know if this was requested (or impossible to make) already I couldn't find anything and YouTube music is very known so sorry if it is.~~

### ​

- [X] I have verified that this request is indeed original

**Comment by Rairof** [comment]
This along with some sort of listen along implementation like Spotify listen along would be great!

**Comment by amsryq** [comment]
There's no official API to fetch what you're listening to on YouTube Music. The closest you can get is with the Last.fm plugin that fetches from your Last.fm profile instead

**Comment by mudrhiod** [comment]
Damn still? I've seen lots of people mention this online when I was reading on it yesterday. It's really silly. Is there no other way of doing like how last FM mobile scrobble apps do it (ik nothing about this idk if this even makes sense)

**Comment by Rairof** [comment]
Hopefully someone knowledgeable enough can find a way around this or just make it from scratch since Google won't do it

**Comment by mudrhiod** [comment]
Hopefully ig

**Comment by amsryq** [comment]
No, Discord app lacks the permission to act like a scrobbler, we can't read what you're listening directly

**Comment by mudrhiod** [comment]
That really sucks

**Comment by Rairof** [comment]
Just sending this editted pic here so users can get a visual representation on what we discussed...
![1701613784204](https://github.com/vendetta-mod/ideas/assets/107706560/00006b1c-f5f1-4d3d-a70d-b190b53d91bd)

**Comment by mollyvita** [comment]
> last FM mobile

just checks for anything using APIs required to play any kind of content like video/music and sends data. it might be possible, but i know nothing about development except python and c#

**Comment by mudrhiod** [comment]
Hm? If it's possible I really hope someone attempts, I find it very crappy how there's not much stuff for mobile rich presences lol. Ik theres things like premid that can show of YTM, wished mobile has more things.

**Comment by amsryq** [comment]
Again, 
> we can't read what you're listening directly.

This idea is impossible as a Vendetta plugin (without external services help) as there's no Spotify Web API equivalent for YTM and we cannot directly retrieve info about the music you are playing on your device. The closest existing solution is to use the Last.fm plugin, and even if an alternative were developed, it would still rely on external services.
